ivi-input-modules/ivi-input-controller/src/ivi-input-controller.c

```diff
@@ -1304,6 +1316,29 @@ input_controller_destroy(struct wl_listener *listener, void *data)
     }
 }
 
+static void
+get_config( struct input_context *ctx)
+{
+    struct weston_config *config = NULL;
+    struct weston_config_section *section = NULL;
+    ctx->seat_default_name = NULL;
+
+    config = wet_get_config(ctx->ivishell->compositor);
+    if (!config)
+        goto exit;
+
+    section = weston_config_get_section(config, "ivi-shell", NULL, NULL);
+    if (!section)
+        goto exit;
+
+    weston_config_section_get_string(section,
```

Comment:
please check how weston is copying the strings to avoid the seat-name configuration will be to long

Reply:
I looked to weston, then see that:

- ivi_input calls weston_config_section_get_string() to get value of key="default-seat", then copy the value. 
- Weston looked key at section->entry_list, then returns corresponding value 
- An entry was created via reading a line in config file, only read 512 bytes per line: [shared/config-parser.c · main · wayland / weston · GitLab](https://gitlab.freedesktop.org/wayland/weston/-/blob/main/shared/config-parser.c#L419) 

=> Weston is preventing for reading a very long string.

Reply:
okey we don't need to limit the size 
but please add the null pointer check for return value of strdup() function

Reply:
I updated the comment, the `get_config()` will return a failure if it can't initialize the default seat name.